src/components/FlyToMeButton.js

```diff
@@ -0,0 +1,32 @@
+import { useContext } from 'react';
+import { MapCenterContext } from '../context/MapCenterContext';
+
+const FlyToMeButton = ({ disabled = false }) => {
+  const valueCenterMap = useContext(MapCenterContext);
+  const { userLocationShared, userCenterMap, setNewCenterMap } = valueCenterMap;
+
+  const isZeroPosition = (position) => {
+    const { lat, lng } = position;
+    return lat === 0 && lng === 0;
+  };
+
+  const handleFlyToMe = () => {
+    if (userLocationShared) {
+      if (!isZeroPosition(userCenterMap)) {
+        setNewCenterMap(userCenterMap);
+      } else {
+        console.warn("The user location doesn't valid (0, 0) position.");
```

Comment:
The console.warn and info are interesting, how do I test this, I'm not sure how to create this error.  Does the console.info message show up in the console if you've shared your location but then turned off sharing and clicked on the "fly to me" button?

Reply:
@meganmckissack : Thanks! A way that you can help you to see those messages is you can use the React Dev tools.

- `The user location doesn't valid (0, 0) position.`: First, you can share your location because the 'Fly to me' button is disabled. Next, you can open your browser tools, go to the React Dev tools and change 'manually' the userCenterMap state to (latitude: 0, longitude: 0), and click on the 'fly to me button'.

![image](https://user-images.githubusercontent.com/8689897/128613863-208b509d-a534-482f-b806-8221b1505d68.png)

- "You didn't share your location.". I put it as defensive programming. But in this case, we don't need to control if we shared our location or not, because if you don't share your location ("Share my location" button), this button will be disabled.  I believe we can remove `if (userLocationShared)`. Thanks!